.github/workflows/test-build-publish.yml

```diff
@@ -38,12 +40,14 @@ jobs:
           uses: actions/setup-python@v2
           with:
             python-version: 3.7
+        - name: Install dependencies
+          run: python -m pip install --upgrade pip build
         - name: Generate Lark Parser
           run: |
             pip install -r requirements.pip -e .
             python hcl2/parser.py
         - name: Build tarball
-          run: python setup.py sdist
+          run: python3 -m build
         - name: Publish to Test PyPI
           uses: pypa/gh-action-pypi-publish@master
           with:
```

Comment:
We might need to split GitHub actions into 2 different files
1. First one for PR checks
2. Second for publishing on release